AWS ingest and monitoring doc redesign (#3684)
elastic/observability-docs: docs/en/observability/cloud-monitoring/aws/monitor-aws-esf.asciidoc

1. Go to *Integrations* in {kib} and search for AWS. 
2. Click the AWS integration, select *Settings* and click *Install AWS assets* to install all the AWS integration assets.

*TBD* == we might need extra details here

[discrete]
[[esf-step-five]]
=== Step 5: Create a new S3 bucket

You create a new S3 bucket and queue for the access logs, then configure the older S3 bucket to generate access logs.

1. In the https://s3.console.aws.amazon.com/s3[AWS S3 console], click *Create* bucket. Give the bucket a name and specify the region where you want it deployed.
+
IMPORTANT: Make sure you create the S3 bucket and SQS queue (next step) in the same region as the bucket containing VPC flow logs.

2. Follow the steps you learned earlier to create an SQS queue and edit the access policy (use the ARNs of the new S3 bucket and queue). Make a note of the queue URL because you will need it later when you configure S3 access log collection.

3. Configure the new S3 bucket to send notifications to the new queue when objects are created (follow the steps you learned earlier).

> 4. Go back to the old S3 bucket (the one that contains VPC flow logs), and under Properties, edit the Server access logging properties. Enable server access logging, and select the new bucket you created as the target bucket.

Reviewer: To set up ESF for the first time, users have to perform two actions:

1. Create an S3 bucket to store the configuration file.
2. Create the actual configuration file.

We can keep these two actions in steps 5 and 6 in this new guide, but we need to update the content.

I can share the steps here or discuss them over Zoom and update the PR afterward.

Author: Thanks for your help, @zmoog 
I scheduled a time slot for us to fix this.

Author: @zmoog After the meeting we had yesterday, this section has been completely rewritten.

[[monitor-aws-esf]]
== Monitor Amazon Web Services ({aws}) with Elastic Serverless Forwarder

++++
<titleabbrev>Monitor {aws} with Elastic Serverless Forwarder</titleabbrev>
++++

> The Elastic Serverless Forwarder (ESF) is an Amazon Web Services (AWS) Lambda function that ships logs from your AWS environment to Elastic. Elastic Serverless Forwarder is published in the AWS Serverless Application Repository (SAR). For more information on ESF, check the https://www.elastic.co/guide/en/esf/current/aws-elastic-serverless-forwarder.html[Elastic Serverless Forwarder Guide]. 

Reviewer 2: ```suggestion
The Elastic Serverless Forwarder (ESF) is an Amazon Web Services (AWS) Lambda function that ships logs from your AWS environment to Elastic. Elastic Serverless Forwarder is published in the AWS Serverless Application Repository (SAR). For more information on ESF, check the {esf-ref}/aws-elastic-serverless-forwarder.html[Elastic Serverless Forwarder Guide]. 
```

- Install and configure the Elastic AWS integration from {kib}
- Visualize and analyze AWS logs in the Elastic Stack

[discrete]
[[aws-esf-prerequisites]]
=== Before you begin

Create a deployment using our hosted {ess} on {ess-trial}[{ecloud}].
The deployment includes an {es} cluster for storing and searching your data, and {kib} for visualizing and managing your data. Elastic Serverless Forwarder works with Elastic Stack 7.17 and later.
You also need an AWS account with permissions to pull the necessary data from AWS.

[discrete]
[[esf-step-one]]
=== Step 1: Create an S3 Bucket

. In the https://s3.console.aws.amazon.com/s3[AWS S3 console], choose *Create bucket* from the left navigation pane. 
. Specify the AWS region in which you want it deployed.
. Enter the bucket name.

For more details, refer to the Amazon documentation on how to https://docs.aws.amazon.com/AmazonS3/latest/userguide/creating-bucket.html[Create your first S3 bucket].

[discrete]
[[esf-step-two]]
=== Step 2: Enable AWS VPC flow logs to be sent to your S3 bucket

1. In the https://console.aws.amazon.com/ec2/[Amazon EC2 console], choose *Network Interfaces* from the left navigation pane.

> 2. Select the the network interface you want to use.

Reviewer 2: ```suggestion
2. Select the network interface you want to use.
```

3. From the *Actions* drop-down menu, choose *Create flow log*.
4. For *Destination*, select *Send to an S3 bucket*.

> 5. For *S3 bucket ARN*, entre the name of the S3 bucket you created in the previous step. 

Reviewer 2: ```suggestion
5. For *S3 bucket ARN*, enter the name of the S3 bucket you created in the previous step. 
```

[discrete]
[[esf-step-six]]
=== Step 6: Create a configuration file to specify the source and destination  

Elastic Serverless Forwarder uses the configuration file to know the input source and the Elastic connection for the destination information.

. In Elastic Cloud, from the AWS Integrations page click *Connection details* on the upper right corner and copy your Cloud ID. 
. Create an encoded API key for authentication. 

You are going to reference both the Cloud ID and the newly created API key from the configuration file. Here is an example:

[source,yml]
----
inputs:
  - type: "s3-sqs"
    id: "<your-sqs-queue-arn>"
    outputs:
      - type: "elasticsearch"
        args:
          cloud_id: "<your-cloud-id>"
          api_key: “<your-api-key>>"
----

. Upload the configuration file you have just created to the S3 bucket you created at step 5.

[discrete]
[[esf-step-seven]]
=== Step 7: Ingest VPC flow logs into Elastic

Deploy the Elastic Serverless Forwarder from AWS SAR and provide appropriate configurations for the Lambda function to start ingesting VPC flow logs into Elastic.

. From the Lambda console select *Applications* and click *Create Application*.

> . From the *Serverless application*, sleect *elastic-serverless-forwarder*.

Reviewer 2: ```suggestion
. From the *Serverless application*, select *elastic-serverless-forwarder*.
```

. On the *Review, configure and deploy* page, fill in the following fields:
+
- *ElasticServerlessForwarderS3Buckets*: Specify the ARN of the S3 Bucket you created at step 1 where the VPC Flow Logs are sent.
+

> - *ElasticServerlessForwarderS3ConfigFile*: Specify the URL of configuration file in the format "s3://bucket-name/config-file-name".

Reviewer 2: ```suggestion
- *ElasticServerlessForwarderS3ConfigFile*: Specify the URL of the configuration file in the format "s3://bucket-name/config-file-name".
```

+

> - *ElasticServerlessForwarderS3SQSEvents*: Specify the S3 SQS Notifications queue used as the trigger of the Lambda function. The value is the ARN of the SQS Queue you created on step 3.

Reviewer 2: Inconsistency nit: On this line, you say "on step 3". On line 146, you say "at step 1".

For more details, refer to the Amazon documentation on how to https://docs.aws.amazon.com/AmazonS3/latest/userguide/creating-bucket.html[Create your first S3 bucket].

[discrete]
[[esf-step-two]]
=== Step 2: Enable AWS VPC flow logs to be sent to your S3 bucket

1. In the https://console.aws.amazon.com/ec2/[Amazon EC2 console], choose *Network Interfaces* from the left navigation pane.
2. Select the the network interface you want to use.
3. From the *Actions* drop-down menu, choose *Create flow log*.
4. For *Destination*, select *Send to an S3 bucket*.
5. For *S3 bucket ARN*, entre the name of the S3 bucket you created in the previous step. 

For more details, refer to the Amazon documentation on how to https://docs.aws.amazon.com/vpc/latest/userguide/flow-logs-s3.html[Create a flow log that publishes to Amazon S3].

[discrete]
[[esf-step-three]]
=== Step 3: Create an SQS queue and notifications for VPC flow logs

The Amazon Simple Queue Service (SQS) event notification on Amazon S3 serves as a trigger for the Lambda function. When a new log file gets written to an Amazon S3 bucket and meets the criteria, a notification is generated that triggers the Lambda function.

> . In the https://console.aws.amazon.com/sqs/[SQS console] create a standard SQS queue that uses the default settings.

Author: I updated the docs by adding the following callout:
`IMPORTANT: Make sure you set a visibility timeout of 910 seconds for any SQS queues you want to use as a trigger. This is 10 seconds greater than the Elastic Serverless Forwarder Lambda timeout. If this requirement is not met, CloudFormation will throw an error.`
